### PR TITLE
Total Cost of contracts per hour and month

### PR DESCRIPTION
**Pull request by AlaaElattar**
### Description
- Showing total cost per hour and month

### Changes

- Add alert at the end of table showing total cost per hour or month.
- If there's no contracts, the alert won't appear

### Related Issues

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1359

Screenshot from 2023-11-09 11-34-39

### CODE REVIEW

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 1]
```diff
@@ -76,6 +76,11 @@
         </v-tooltip>
       </template>
     </ListTable>
+    <div class="pt-4">
+      <v-alert variant="tonal" color="secondary" class="pt-4" v-if="contracts.length !== 0"
+        >Total Cost: {{ totalCost }} TFT/hour ≈ {{ (totalCost * 24 * 30).toFixed(3) }} TFT/month</v-alert
```
Comment (amiraabouhadid) [inline packages/playground/src/weblets/tf_contracts_list.vue:81]: you can just do `v-if="contracts.length"`

**Comment by amiraabouhadid** [comment]
is it supposed to give 0 here? there are many contracts but the cost displayed is 0. 
image

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]

**Comment by AlaaElattar** [comment]
> is it supposed to give 0 here? there are many contracts but the cost displayed is 0. image

I can check also if the `totalCost` is zero so i won't show the dialog. what do u think of this ?

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 2]
```diff
@@ -306,6 +314,18 @@ function getNodeStateColor(state: NodeStatus): string {
...
   }
 }
+
+function getTotalCost(contracts: NormalizedContract[]) {
+  totalCost.value = 0;
+  for (const contract of contracts) {
+    const matching = contract.consumption.match(/(\d+(\.\d+)?) TFT\/hour/);
+    if (matching) {
+      const value = parseFloat(matching[1]);
+      totalCost.value += value;
```
Comment (AhmedHanafy725) [inline packages/playground/src/weblets/tf_contracts_list.vue:324]: please use `Decimal` in any money calculations

**Review by zaelgohary — APPROVED** [review]

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 3]
```diff
@@ -306,6 +315,18 @@ function getNodeStateColor(state: NodeStatus): string {
...
   }
 }
+
+function getTotalCost(contracts: NormalizedContract[]) {
+  totalCost.value = 0;
+  for (const contract of contracts) {
+    const matching = contract.consumption.match(/(\d+(\.\d+)?) TFT\/hour/);
+    if (matching) {
+      const value = new Decimal(matching[1]);
+      totalCost.value += +value;
```
[the remark this answers is not in the document]
Comment (AlaaElattar) [reply packages/playground/src/weblets/tf_contracts_list.vue:325]: so u want `totalCost` to be of type `Decimal` ?

[the remark this answers is not in the document]
Comment (AhmedHanafy725) [reply packages/playground/src/weblets/tf_contracts_list.vue:325]: no, you can do the calculation with Decimal and then convert it to a number at the end to show it to the user

**Review by AhmedHanafy725 — APPROVED** [review]

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 4]
```diff
@@ -306,6 +315,18 @@ function getNodeStateColor(state: NodeStatus): string {
       return "warning";
   }
 }
+
+function getTotalCost(contracts: NormalizedContract[]) {
```
Comment (xmonader) [inline packages/playground/src/weblets/tf_contracts_list.vue:319]: I believe we should introduce the raw field instead along with the consumption string, so field `tftPerHour` on the normalized contract object, and drop the regex part

**Review by xmonader — CHANGES_REQUESTED** [review]
Added a change request, let me know if you guys still want to proceed with the regular expression and why.

**Review by xmonader — APPROVED** [review]
Looks much cleaner now 💯 Thank you

**Review by amiraabouhadid — APPROVED** [review]

**Pull request merged**